### PR TITLE
Empty name not allowed in theming app

### LINKED ISSUE #4858: Do not allow empty name in theming app
If the name is blank in the admin theming section, the slogan should not be preceded by a hyphen

### PR DESCRIPTION
**Pull request by sagorika1996**
Fixes #4858
screenshot from 2018-02-21 18-27-30

### CODE REVIEW

**Comment by nickvergessen** [comment]
Can you please sign your commits? See https://github.com/nextcloud/server/blob/master/CONTRIBUTING.md#sign-your-work for more information.

You can do this manually for now:
```
git commit --amend -s
git push --force origin <your branch>
```

File: apps/theming/js/settings-admin.js [hunk 1]
```diff
@@ -175,13 +175,40 @@ $(document).ready(function () {
 		$('#upload-login-background').click();
 	});
 
+	function checkName () {
+		var length = $('#theming-name').val().length;
+		try {
+			if (length > 0) {
+				return true;
+			} else {
+				throw 'Name cannot be empty';
```
Comment (nickvergessen) [inline apps/theming/js/settings-admin.js]: Should use `t('theming', 'Name cannot be empty')` so the string is translated.

**Comment by MorrisJobke** [comment]
cc @nextcloud/theming

File: apps/theming/js/settings-admin.js (hunk 1 above)
Comment (sagorika1996) [reply apps/theming/js/settings-admin.js]: Done.

**Comment by juliushaertl** [comment]
@sagorika1996 Code looks good. Thanks for your contribution @sagorika1996 
Can you also add the signoff-message to the latest commit? Then it is fine from my side.

**Comment by sagorika1996** [comment]
@juliushaertl Oops, missed that. Added the signoff-message.

**Comment by skjnldsv** [comment]
@sagorika1996 you need to squash your unsigned commits or edit them. All of them need to be signed :)

**Comment by juliushaertl** [comment]
@sagorika1996 The following steps should help you to get a clean commit history and add the sign off message to the last commit (where it is missing):

- Fetch the latest changes from the remote repo `git fetch`
- Rebase your changes to the latest master `git rebase origin/master`
- Edit apps/theming/js/settings-admin.js to get rid of the merge conflict
From the coparison just keep the fixed line with the call to t()
```
++<<<<<<< HEAD
 +                              throw t('theming', 'Name cannot be empty');
++=======
+                               throw 'Name cannot be empty';
++>>>>>>> Empty name not allowed
```
- Add the changes using `git add apps/theming/js/settings-admin.js`
- Continue with the rebase `git rebase --continue`
- Skip the commit that seems to be duplicate in the history `git rebase --skip`
- Run `git commit --amend --signoff` to add the sign off message to the last commit
- Force push the changes `git push -f`

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Review by rullzer — APPROVED** [review]

**Comment by rullzer** [comment]
THNX @sagorika1996 

I took the liberty of squashing your commits.
I also invited you to the nextcloud org. So you can create branches directly in our repo's so other people can more easily collaborate with you.

**Review by juliusknorr — APPROVED** [review]

**Comment by sagorika1996** [comment]
@rullzer thanks for the help and the invitation :)

**Pull request merged**